client/query.go

```diff
@@ -77,7 +77,7 @@ func (ctx Context) queryABCI(req abci.RequestQuery) (abci.ResponseQuery, error)
 	}
 
 	opts := rpcclient.ABCIQueryOptions{
-		Height: ctx.Height,
+		Height: req.Height,
```

Comment:
In CLI (with --height) and grpc queries (with the height header), iirc we populate the client ctx.Height. So I think we should use req.Height if that's set, or fallback to ctx.Height.

I believe that's the cause of all failing tests

Reply:
That makes sense. Is there a reason we don't populate the `req` in CLI and API pathways, instead of `ctx`? Just curious.

Reply:
> Is there a reason we don't populate the req in CLI and API pathways

Yeah that could work too, where do you see this logic happening?

Reply:
So `Context#queryABCI` gets call from 2-3 places, one of which doesn't have a height to provide, so we cannot solely rely on `req.Height`.

I would do two things:

1. Update `RunGRPCQuery` to set `Height` in `abciReq`.
2. Update your logic to fallback on `ctx.Height` if `req.Height` is zero.